zebra-network/src/protocol/external/codec.rs

```diff
@@ -514,8 +509,12 @@ impl Codec {
         })
     }
 
+    /// Deserialize an `addr` (v1) message into a list of `MetaAddr`s.
     fn read_addr<R: Read>(&self, reader: R) -> Result<Message, Error> {
-        Ok(Message::Addr(Vec::zcash_deserialize(reader)?))
+        let addrs: Vec<AddrV1> = reader.zcash_deserialize_into()?;
+
+        let addrs = addrs.into_iter().map(Into::into).collect();
+        Ok(Message::Addr(addrs))
```

Comment:
Nit: Maybe use a different name to avoid ambiguity/confusion?

```suggestion
        let meta_addrs = addrs.into_iter().map(Into::into).collect();
        Ok(Message::Addr(meta_addrs))
```

Reply:
I am deliberately shadowing the variable name, so it doesn't accidentally get re-used.
The type system ensures that `addrs` is the correct type for a `Message::Addr`.

```suggestion
        // Convert the received address format to Zebra's internal `MetaAddr`.
        let addrs = addrs.into_iter().map(Into::into).collect();
        Ok(Message::Addr(addrs))
```